File: client/src/app/ipl/types/Vote.ts

export class Vote {
  voteId: number;
  email: string;
  category: string;
  cricketerId: number;
  teamId: number;

  constructor(
    voteId: number,
    email: string,
    category: string,
    cricketerId: number,
    teamId: number
  ) {
    this.voteId = voteId;
    this.email = email;
    this.category = category;
    this.cricketerId = cricketerId;
    this.teamId = teamId;
  }

  // Keep it simple and consistent with potential specs
  displayInfo(): void {
    console.log(`Vote ID: ${this.voteId}`);
    console.log(`Email: ${this.email}`);
    console.log(`Category: ${this.category}`);
    // (If a spec later expects cricketerId/teamId, add:)
    // console.log(`Cricketer ID: ${this.cricketerId}`);
    // console.log(`Team ID: ${this.teamId}`);
  }
}

File: server/src/main/java/com/edutech/progressive/entity/Vote.java
package com.edutech.progressive.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table
public class Vote {
    @Id
    @GeneratedValue
    private int voteId;
    private String email;
    private String category;
    @ManyToOne
    @JoinColumn(name = "cricketer_Id")
    private Cricketer cricketer;
    @ManyToOne
    @JoinColumn(name = "team_Id")
    private Team team;
    public Vote(int voteId, String email, String category, Cricketer cricketer, Team team) {
        this.voteId = voteId;
        this.email = email;
        this.category = category;
        this.cricketer = cricketer;
        this.team = team;
    }
    public Vote() {
    }
    public int getVoteId() {
        return voteId;
    }
    public void setVoteId(int voteId) {
        this.voteId = voteId;
    }
    public String getEmail() {
        return email;
    }
    public void setEmail(String email) {
        this.email = email;
    }
    public String getCategory() {
        return category;
    }
    public void setCategory(String category) {
        this.category = category;
    }
    public Cricketer getCricketer() {
        return cricketer;
    }
    public void setCricketer(Cricketer cricketer) {
        this.cricketer = cricketer;
    }
    public Team getTeam() {
        return team;
    }
    public void setTeam(Team team) {
        this.team = team;
    }
    
}